Suggestion for adding a type parameter happens a little too often.

The following code now emits a suggestion to add a type parameter (since #68447 in 1.42):

```rust
pub struct S {
    pub f: Option<PathBuf>
}
```

Emits:

```
error[E0412]: cannot find type `PathBuf` in this scope
 --> src/lib.rs:2:19
  |
2 |     pub f: Option<PathBuf>
  |                   ^^^^^^^ not found in this scope
  |
help: possible candidate is found in another module, you can import it into scope
  |
1 | use std::path::PathBuf;
  |
help: you might be missing a type parameter
  |
1 | pub struct S<PathBuf> {
  |             ^^^^^^^^^
```

The suggestion to add `<PathBuf>` is clearly wrong (in my eyes), and I feel like I am seeing this suggestion a lot.

Perhaps, if there is already a "possible candidate import" suggestion, it could skip the suggestion to add a type parameter?  Or maybe some other heuristic to dial back how often it shows up?

(BTW, it is a great suggestion when it applies, but more often than not I'm getting a wrong hint.)

--- a/src/librustc_resolve/lib.rs
+++ b/src/librustc_resolve/lib.rs
@@ -2480,8 +2480,7 @@ impl<'a> Resolver<'a> {
             let (span, found_use) = UsePlacementFinder::check(krate, node_id);
             if !candidates.is_empty() {
                 diagnostics::show_candidates(&mut err, span, &candidates, better, found_use);
-            }
-            if let Some((span, msg, sugg, appl)) = suggestion {
+            } else if let Some((span, msg, sugg, appl)) = suggestion {
                 err.span_suggestion(span, msg, sugg, appl);
             }
             err.emit();

--- a/src/test/ui/privacy/privacy-ns1.stderr
+++ b/src/test/ui/privacy/privacy-ns1.stderr
@@ -63,10 +63,6 @@ LL | use foo2::Bar;
    |
 LL | use foo3::Bar;
    |
-help: you might be missing a type parameter
-   |
-LL | fn test_glob3<Bar>() {
-   |              ^^^^^
 
 error[E0107]: wrong number of const arguments: expected 0, found 1
   --> $DIR/privacy-ns1.rs:35:17

--- a/src/test/ui/suggestions/no-extern-crate-in-type.stderr
+++ b/src/test/ui/suggestions/no-extern-crate-in-type.stderr
@@ -8,10 +8,6 @@ help: possible candidate is found in another module, you can import it into scop
    |
 LL | use foo::Foo;
    |
-help: you might be missing a type parameter
-   |
-LL | type Output<Foo> = Option<Foo>;
-   |            ^^^^^
 
 error: aborting due to previous error
 

